### PR TITLE
Move taglib core date logic to a bean

### PR DESCRIPTION
**Pull request by jameskleeh**
I left the tests in there but commented out for when we move to requiring Java 8. At that time we can just uncomment them.

### CODE REVIEW

File: grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/DefaultGrailsTagDateHelper.groovy [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package org.grails.plugins.web
```
Comment (graemerocher) [inline grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/DefaultGrailsTagDateHelper.groovy]: license

File: grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/DefaultGrailsTagDateHelper.groovy [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import org.apache.commons.lang.time.FastDateFormat
+
+
+class DefaultGrailsTagDateHelper implements GrailsTagDateHelper {
```
Comment (graemerocher) [inline grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/DefaultGrailsTagDateHelper.groovy]: groovydoc please (author, version, description)

File: grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/GrailsTagDateHelper.groovy [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package org.grails.plugins.web
+
+/**
+ * Created by jameskleeh on 10/5/16.
+ */
+interface GrailsTagDateHelper {
```
Comment (graemerocher) [inline grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/GrailsTagDateHelper.groovy]: groovydoc please (author, version, description)

File: grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/GrailsTagDateHelper.groovy [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package org.grails.plugins.web
```
Comment (graemerocher) [inline grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/GrailsTagDateHelper.groovy]: license

File: grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/GrailsTagDateHelper.groovy (hunk 3 above)
Comment (graemerocher) [inline grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/GrailsTagDateHelper.groovy]: Also groovydoc form for methods

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/DefaultGrailsTagDateHelper.groovy (hunk 1 above)
Comment (jameskleeh) [inline grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/DefaultGrailsTagDateHelper.groovy]: Done

File: grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/GrailsTagDateHelper.groovy (hunk 4 above)
Comment (jameskleeh) [inline grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/GrailsTagDateHelper.groovy]: Done

**Pull request merged**